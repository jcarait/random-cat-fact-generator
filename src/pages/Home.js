import { Navigate } from "react-router-dom";
import Button from "../components/button/button";
import Card from "../components/card/card";
import catImage from "../images/cat.png";

export default function Home({ props, onClick }) {
  const { data, isLoading, error } = props;

  return (
    <div className="App">
      <header className="App-header">
        <h1 className="title">Random Cat Fact Generator</h1>
        {isLoading && <p data-testid="loading">Loading...</p>}
        <div className="container">
          <img
            className="cat-image"
            src={catImage}
            alt="a cat raising its paw up"
          ></img>
          <Card data={!isLoading && data.fact} />
        </div>
        <Button label="refresh" onClick={onClick} />
      </header>
    </div>
  );
}
