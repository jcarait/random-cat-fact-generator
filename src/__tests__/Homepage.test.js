import renderer from "react-test-renderer";
import Home from "../pages/Home";

describe("<HomePage />", () => {
  it("renders correctly", () => {
    const data = {
      fact: "Random cat fact",
    };
    const loading = false;
    const error = "";

    const tree = renderer
      .create(<Home props={{ data, loading, error }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
