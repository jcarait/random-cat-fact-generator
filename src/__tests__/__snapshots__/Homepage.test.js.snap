// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="title"
    >
      Random Cat Fact Generator
    </h1>
    <div
      className="container"
    >
      <img
        alt="a cat raising its paw up"
        className="cat-image"
        src="cat.png"
      />
      <div
        className="card"
        data-testid="card"
      >
        <p>
          Random cat fact
        </p>
      </div>
    </div>
    <div
      className="button"
    >
      refresh
    </div>
  </header>
</div>
`;
